src/native-common/TextInput.tsx

```diff
@@ -48,51 +48,60 @@ export class TextInput extends React.Component<Types.TextInputProps, TextInputSt
         }
     }
 
-    render() {
-        const editable = (this.props.editable !== undefined ? this.props.editable : true);
-        const blurOnSubmit = this.props.blurOnSubmit || !this.props.multiline;
+    protected _render(props: RN.TextInputProps): JSX.Element {
         return (
             <RN.TextInput
-                ref={ 'nativeTextInput' }
-                multiline={ this.props.multiline }
-                style={ Styles.combine([_styles.defaultTextInput, this.props.style]) }
-                value={ this.state.inputValue }
-
-                autoCorrect={ this.props.autoCorrect }
-                spellCheck={ this.props.spellCheck }
-                autoCapitalize={ this.props.autoCapitalize }
-                autoFocus= { this.props.autoFocus }
-                keyboardType={ this.props.keyboardType }
-                editable={ editable }
-                selectionColor={ this.props.selectionColor }
-                maxLength={ this.props.maxLength }
-                placeholder={ this.props.placeholder }
-                defaultValue={ this.props.value }
-                placeholderTextColor={ this.props.placeholderTextColor }
-                onSubmitEditing={this.props.onSubmitEditing }
-                onKeyPress={ this._onKeyPress as any }
-                onChangeText={ this._onChangeText }
-                onSelectionChange={ this._onSelectionChange as any }
-                onFocus={ this._onFocus }
-                onBlur={ this._onBlur }
-                onScroll={ this._onScroll }
-                selection={{ start: this._selectionStart, end: this._selectionEnd }}
-                secureTextEntry={ this.props.secureTextEntry }
-
-                textAlign={ this.props.textAlign }
-                keyboardAppearance={ this.props.keyboardAppearance }
-                returnKeyType={ this.props.returnKeyType }
-                disableFullscreenUI={ this.props.disableFullscreenUI }
-                blurOnSubmit={ blurOnSubmit }
-                textBreakStrategy={ 'simple' }
-                accessibilityLabel={ this.props.accessibilityLabel }
-                allowFontScaling={ this.props.allowFontScaling }
-                maxContentSizeMultiplier={ this.props.maxContentSizeMultiplier }
-                underlineColorAndroid='transparent'
+                {...props}
             />
         );
     }
 
+    render() {
+        const editable = (this.props.editable !== undefined ? this.props.editable : true);
+        const blurOnSubmit = this.props.blurOnSubmit || !this.props.multiline;
+
+        const internalProps: RN.TextInputProps = {
+            ref: 'nativeTextInput',
```

Comment:
Refs by string are deprecated. And it makes sense to upgrade the previously existing refs by string to callbacks in some files above too, since you're touching the files anyway.

Reply:
Ok. I knew this will come..

---
In reply to: [159245310](https://github.com/Microsoft/reactxp/pull/404#discussion_r159245310) [](ancestors = 159245310)